match/lib/match/storage/s3_storage.rb

```diff
@@ -102,6 +102,10 @@ def download
         self.working_directory = Dir.mktmpdir
 
         s3_client.find_bucket!(s3_bucket).objects(prefix: s3_object_prefix).each do |object|
+
+          # prevent download if file_path is a dir
+          next if object.key.end_with?("/")
```

Comment:
Why not `File.directory?(object.key)`?
I don't know about specifics of S3, but a file path can be a directory and not end with `/`

Reply:
Ah, gotcha. The strings represent remote directories, not local ones, so the system can't tell if the string is a dir or not 😅 

Maybe we could document this, WDYT?

```suggestion
          # Prevent download if the file path is a directory.
          # We need to check if string ends with "/" instead of using `File.directory?` because
          # the string represent a remote location, not a local file in disk.
          next if object.key.end_with?("/")
```

Reply:
You're right, it's not obvious why it happens. Added your comment. Thanks